sklearn/datasets/_base.py

```diff
@@ -1119,16 +1175,16 @@ def load_sample_images():
     # import PIL only when needed
     from ..externals._pilutil import imread
 
-    module_path = join(dirname(__file__), "images")
-    with open(join(module_path, "README.txt")) as f:
-        descr = f.read()
-    filenames = [
-        join(module_path, filename)
-        for filename in sorted(os.listdir(module_path))
-        if filename.endswith(".jpg")
-    ]
-    # Load image data for each image in the source folder.
-    images = [imread(filename) for filename in filenames]
+    descr = load_descr(IMAGES_MODULE, "README.txt")
+
+    filenames = list()
+    images = list()
+    for filename in sorted(importlib_resources.contents(IMAGES_MODULE)):
```

Comment:
We can directly use `.contents()` (which returns a list of file names and is available since `python 3.7`) to avoid using `.files()`. It was my mistake not remembering the option.